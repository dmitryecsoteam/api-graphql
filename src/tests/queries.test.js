const graphqlServer = require('../server/serverExpress');
const axios = require('axios');

const url = `http://localhost:${process.env.API_PORT}/${process.env.API_CONTEXT_ROOT}`;
const method = 'POST';
const headers = { 'Content-Type': 'application/json' };


jest.mock('../models/Destination.js');
jest.mock('../models/Origin.js');
jest.mock('../models/Travel.js');
jest.mock('../models/User.js');
jest.mock('../auth/JWT.js');


let server;

beforeAll(async () => {
    server = await graphqlServer.init();
    //await mongo.connect();
});

afterAll(async () => {
    await server.close();
    console.log("Server stopped");
});

test('should respond to originStartsWith query with lowercase letter', async () => {

    const data = JSON.stringify({ query: '{originStartsWith(name: "t") { _id nameEn name iata countryEn } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should respond to destinationStartsWith query with lowercase letter', async () => {

    const data = JSON.stringify({ query: '{destinationStartsWith(name: "o") { _id nameEn name iata countryEn } }' })

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should respond to destinationRating with all parameters passed', async () => {

    const data = JSON.stringify({ query: '{destinationRating(museumRating: 3 beachRating: 3 foodRating: 3 shoppingRating: 3 nightlifeRating: 3 natureRating: 3) { _id name iata nameEn countryEn museumRating museumDescription beachRating beachDescription foodRating foodDescription shoppingRating shoppingDescription nightlifeRating nightlifeDescription natureRating natureDescription } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should respond to destination query', async () => {

    const data = JSON.stringify({ query: '{destination(_id: 3) { _id name iata nameEn countryEn museumRating museumDescription beachRating beachDescription foodRating foodDescription shoppingRating shoppingDescription natureRating natureDescription } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should respond to travel query', async () => {

    const data = JSON.stringify({ query: '{travel(origin: 1, destination: 2, date: "2019-07-07") { _id destination { _id nameEn countryEn } date priceAirplane priceHotel } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should respond to travelFull query', async () => {

    const data = JSON.stringify({ query: '{travelFull(_id: "5bbce95f81e97acfcd8e0a76") { _id destination { _id nameEn countryEn foundingDate } origin { nameEn } date priceAirplane priceHotel } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data).toMatchSnapshot();
});

test('should return current user info', async () => {
    
    const data = JSON.stringify({ query: '{currentUser { email name } }' });

    const response = await axios({ method, url, headers: { ...headers, Authorization: 'Bearer valid_token' }, data });

    expect(response.data).toMatchSnapshot();
});

test('should return null for currentUser without auth header', async () => {

    const data = JSON.stringify({ query: '{currentUser { email name } }' });

    const response = await axios({ method, url, headers, data });

    expect(response.data.data.currentUser).toBeNull;
});

test('should return getNotifications for authorized user', async () => {

    const data = JSON.stringify({ query: '{getNotifications { travelId origin { nameEn } destination { nameEn } date priceAirplaneLast priceHotelLast } }' });

    const response = await axios({ method, url, headers: { ...headers, Authorization: 'Bearer valid_token' }, data });

    expect(response.data).toMatchSnapshot();
});

test('should return "Unauthorized" for query without token', async () => {

    const data = JSON.stringify({ query: '{getNotifications { travelId origin { nameEn } destination { nameEn } date priceAirplaneLast priceHotelLast } }' });

    const response = await axios({ method, url, headers, data });
    
    expect(response.data).toMatchSnapshot();
});