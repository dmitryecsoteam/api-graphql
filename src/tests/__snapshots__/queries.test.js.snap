// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should respond to destination query 1`] = `"{\\"data\\":{\\"destination\\":{\\"_id\\":3,\\"name\\":[\\"Nagoya\\",\\"Нагоя\\",\\"名古屋市\\",\\"NGO\\",\\"NKM\\"],\\"iata\\":\\"NGO\\",\\"nameEn\\":\\"Nagoya\\",\\"countryEn\\":\\"Japan\\",\\"museumRating\\":1,\\"museumDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"beachRating\\":0,\\"beachDescription\\":\\"Lorem ipsum dolor sit amet, consectetur adipiscing elit\\",\\"foodRating\\":2,\\"foodDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"shoppingRating\\":4,\\"shoppingDescription\\":\\"Lorem ipsum dolor sit amet, consectetur adipiscing elit\\",\\"natureRating\\":1,\\"natureDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\"}}}"`;

exports[`should respond to destinationRating with all parameters passed 1`] = `"{\\"data\\":{\\"destinationRating\\":[{\\"_id\\":5,\\"name\\":[\\"Nagasaki\\",\\"Нагасаки\\",\\"長崎市\\",\\"NGS\\"],\\"iata\\":\\"NGS\\",\\"nameEn\\":\\"Nagasaki\\",\\"countryEn\\":\\"Japan\\",\\"museumRating\\":4,\\"museumDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\",\\"beachRating\\":3,\\"beachDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"foodRating\\":4,\\"foodDescription\\":\\"Proin pharetra ante est, eget convallis magna tempor id. Etiam vitae aliquet massa\\",\\"shoppingRating\\":3,\\"shoppingDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"nightlifeRating\\":5,\\"nightlifeDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"natureRating\\":5,\\"natureDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\"}]}}"`;

exports[`should respond to destinationStartsWith query with lowercase letter 1`] = `"{\\"data\\":{\\"destinationStartsWith\\":[{\\"_id\\":1,\\"nameEn\\":\\"Osaka\\",\\"name\\":[\\"Osaka\\",\\"Осака\\",\\"大阪市\\",\\"KIX\\",\\"OSA\\",\\"ITM\\"],\\"iata\\":\\"OSA\\",\\"countryEn\\":\\"Japan\\"},{\\"_id\\":4,\\"nameEn\\":\\"Sapporo\\",\\"name\\":[\\"Sapporo\\",\\"Саппоро\\",\\"札幌市\\",\\"SPK\\",\\"CTS\\",\\"OKD\\"],\\"iata\\":\\"SPK\\",\\"countryEn\\":\\"Japan\\"}]}}"`;

exports[`should respond to originStartsWith query with lowercase letter 1`] = `"{\\"data\\":{\\"originStartsWith\\":[{\\"_id\\":1,\\"nameEn\\":\\"Tokyo\\",\\"name\\":[\\"Tokyo\\",\\"東京\\",\\"Токио\\",\\"TYO\\",\\"NRT\\",\\"HND\\"],\\"iata\\":\\"TYO\\",\\"countryEn\\":\\"Japan\\"}]}}"`;

exports[`should respond to travel query 1`] = `"{\\"data\\":{\\"travel\\":{\\"_id\\":\\"5bbce95f81e97acfcd8e0a76\\",\\"destination\\":{\\"_id\\":1,\\"nameEn\\":\\"Osaka\\",\\"countryEn\\":\\"Japan\\"},\\"date\\":\\"2018-10-24\\",\\"priceAirplane\\":53,\\"priceHotel\\":75}}}"`;

exports[`should respond to travelFull query 1`] = `"{\\"data\\":{\\"travelFull\\":{\\"_id\\":\\"5bbce95f81e97acfcd8e0a76\\",\\"destination\\":{\\"_id\\":1,\\"nameEn\\":\\"Osaka\\",\\"countryEn\\":\\"Japan\\"},\\"date\\":\\"2018-10-24\\",\\"priceAirplane\\":53,\\"priceHotel\\":25}}}"`;
